src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,299 @@
+/**
+ * Calls an array of actions based on the value of o->oAction
+ */
 void bhv_fish_loop(void) {
     obj_call_action_function(sFishActions);
 }
 
-void func_802BF59C(s32 a0) {
-    f32 sp1C = o->parentObj->oPosY;
-    if (gCurrLevelNum == LEVEL_SA) {
-        if (500.0f < absf(o->oPosY - o->oFishGroupUnkF8))
-            a0 = 10;
-        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishGroupUnkF8, a0);
-    } else if (sp1C - 100.0f - o->oFishGroupUnk10C < o->oPosY
-               && o->oPosY < sp1C + 1000.0f + o->oFishGroupUnk10C)
-        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishGroupUnkF8, a0);
+/**
+ * Move object up or down towards oFishPosY by the speed variable
+ * In Secret Aquarium, the speed variable is always ten.
+ */
+void fish_regroup(s32 speed) {
+    // Set pPosY to the parentY.
+    f32 pPosY = o->parentObj->oPosY;
+    
+    // If level is SA and fish height minus oPosY is higher than 500 then speed = 10. 
+    if (gCurrLevelNum == 20) {
+        if (500.0f < absf(o->oPosY - o->oFishPosY)) {
+            speed = 10;
+        }
+        // Increase fish Y coord at the incremented speed.
+        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishPosY, speed);
+     /**
+     * For other levels, if oFishRandomYDistance - pPosy - 100.0f is lower than 
+     * o->oPosY - 100 and * o->PosY is lower than the pPosY + 1000.0f +
+     * oFishRandomYDistance then have the fish 
+     * symmetrically approach the target oFishPosY.
+     */
+    } else if (pPosY - 100.0f - o->oFishRandomYDistance < o->oPosY
+               && o->oPosY < pPosY + 1000.0f + o->oFishRandomYDistance) {
+        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishPosY, speed);
+    }
 }
-
-void ActionFishGroup1(void) {
-    f32 sp1C = o->oPosY - gMarioObject->oPosY;
-    if (o->oTimer < 10)
+/**
+ * Moves fish forward at a random velocity and sets a random rotation.
+ */
+void fish_group_act_rotation(void) {
+    f32 fishY = o->oPosY - gMarioObject->oPosY;
+    
+    // Animation accel parameters altered by timer.
+    if (o->oTimer < 10) {
         func_8029ED98(0, 2.0f);
-    else
+    } else {
         func_8029ED98(0, 1.0f);
+    }
+    
+    /**
+     * Assigns oForwardVel, oFishRandomOffset, & oFishRandomDistance to a random floats.
+     */
     if (o->oTimer == 0) {
         o->oForwardVel = RandomFloat() * 2 + 3.0f;
-        if (gCurrLevelNum == LEVEL_SA)
-            o->oFishGroupUnkFC = RandomFloat() * 700.0f;
-        else
-            o->oFishGroupUnkFC = RandomFloat() * 100.0f;
-        o->oFishGroupUnk104 = RandomFloat() * 500 + 200.0f;
+        if (gCurrLevelNum == 20) {
+            o->oFishRandomOffset = RandomFloat() * 700.0f;
+        } else {
+            o->oFishRandomOffset = RandomFloat() * 100.0f;
+        }
+        o->oFishRandomDistance = RandomFloat() * 500 + 200.0f;
     }
-    o->oFishGroupUnkF8 = gMarioObject->oPosY + o->oFishGroupUnkFC;
+    
+    // Rotate fish towards mario at a rate of 5.625 degrees (0x400)
```

Comment:
Capitalize Mario, also only use Hexadecimal Angle Units (HAU) here since the value was pretty clearly not originally in degrees (5.625?). In general only reference degrees if the value was clearly converted from degrees to HAU